src/metalperformanceshadersgraph.cs

```diff
@@ -0,0 +1,1831 @@
+
+		// -(NSArray<MPSGraphTensor *> * _Nonnull)ifWithPredicateTensor:(MPSGraphTensor * _Nonnull)predicateTensor thenBlock:(MPSGraphIfThenElseBlock _Nonnull)thenBlock elseBlock:(MPSGraphIfThenElseBlock _Nullable)elseBlock name:(NSString * _Nullable)name __attribute__((swift_name("if(_:then:else:name:)")));
+		[Export ("ifWithPredicateTensor:thenBlock:elseBlock:name:")]
+		MPSGraphTensor[] If (MPSGraphTensor predicateTensor, MPSGraphIfThenElseBlock thenBlock, [NullAllowed] MPSGraphIfThenElseBlock elseBlock, [NullAllowed] string name);
+
+		// -(NSArray<MPSGraphTensor *> * _Nonnull)whileWithInitialInputs:(NSArray<MPSGraphTensor *> * _Nonnull)initialInputs before:(MPSGraphWhileBeforeBlock _Nonnull)before after:(MPSGraphWhileAfterBlock _Nonnull)after name:(NSString * _Nullable)name __attribute__((swift_name("while(initialInputs:before:after:name:)")));
+		[Export ("whileWithInitialInputs:before:after:name:")]
+		MPSGraphTensor[] While (MPSGraphTensor[] initialInputs, MPSGraphWhileBeforeBlock before, MPSGraphWhileAfterBlock after, [NullAllowed] string name);
+
+		// -(NSArray<MPSGraphTensor *> * _Nonnull)forLoopWithLowerBound:(MPSGraphTensor * _Nonnull)lowerBound upperBound:(MPSGraphTensor * _Nonnull)upperBound step:(MPSGraphTensor * _Nonnull)step initialBodyArguments:(NSArray<MPSGraphTensor *> * _Nonnull)initialBodyArguments body:(MPSGraphForLoopBodyBlock _Nonnull)body name:(NSString * _Nullable)name __attribute__((swift_name("for(lowerBound:upperBound:step:initialBodyArguments:body:name:)")));
+		[Export ("forLoopWithLowerBound:upperBound:step:initialBodyArguments:body:name:")]
+		MPSGraphTensor[] For (MPSGraphTensor lowerBound, MPSGraphTensor upperBound, MPSGraphTensor step, MPSGraphTensor[] initialBodyArguments, MPSGraphForLoopBodyBlock body, [NullAllowed] string name);
+
+		// -(NSArray<MPSGraphTensor *> * _Nonnull)forLoopWithNumberOfIterations:(MPSGraphTensor * _Nonnull)numberOfIterations initialBodyArguments:(NSArray<MPSGraphTensor *> * _Nonnull)initialBodyArguments body:(MPSGraphForLoopBodyBlock _Nonnull)body name:(NSString * _Nullable)name __attribute__((swift_name("for(numberOfIterations:initialBodyArguments:body:name:)")));
+		[Export ("forLoopWithNumberOfIterations:initialBodyArguments:body:name:")]
+		MPSGraphTensor[] For (MPSGraphTensor numberOfIterations, MPSGraphTensor[] initialBodyArguments, MPSGraphForLoopBodyBlock body, [NullAllowed] string name);
+	}
+
+	// typedef NSArray<MPSGraphTensor *> * _Nonnull (^MPSGraphControlFlowDependencyBlock)();
+	delegate MPSGraphTensor[] MPSGraphControlFlowDependencyBlock ();
+
+	// typedef NSArray<MPSGraphTensor *> * _Nonnull (^MPSGraphIfThenElseBlock)();
+	delegate MPSGraphTensor[] MPSGraphIfThenElseBlock ();
+
+	// typedef MPSGraphTensor * _Nonnull (^MPSGraphWhileBeforeBlock)(NSArray<MPSGraphTensor *> * _Nonnull, NSMutableArray<MPSGraphTensor *> * _Nonnull);
+	delegate MPSGraphTensor MPSGraphWhileBeforeBlock (MPSGraphTensor[] arg0, NSMutableArray<MPSGraphTensor> arg1);
```

Comment:
Better parameter names?

```suggestion
	delegate MPSGraphTensor MPSGraphWhileBeforeBlock (MPSGraphTensor[] inputTensors, NSMutableArray<MPSGraphTensor> resultTensors);
```